pkg/compactor/bucket_compactor_e2e_test.go

```diff
@@ -331,22 +333,48 @@ func TestGroupCompactE2E(t *testing.T) {
 			},
 			// Due to TSDB compaction delay (not compacting fresh block), we need one more block to be pushed to trigger compaction.
 			{
-				numSamples: 100, mint: 3000, maxt: 4000, extLset: extLabels2, res: 124,
+				numFloatSamples: 100, mint: 3000, maxt: 4000, extLset: extLabels2, res: 124,
+				series: []labels.Labels{
+					labels.FromStrings("a", "7"),
+				},
+			},
+			// Third group (extLabels3) for native histograms.
+			{
+				numHistogramSamples: 100, mint: 500, maxt: 1000, extLset: extLabels3, res: 124,
+				series: []labels.Labels{
+					labels.FromStrings("a", "1"),
+					labels.FromStrings("a", "2", "b", "2"),
+					labels.FromStrings("a", "3"),
+					labels.FromStrings("a", "4"),
+				},
+			},
+			{
+				numHistogramSamples: 100, mint: 2000, maxt: 3000, extLset: extLabels3, res: 124,
+				series: []labels.Labels{
+					labels.FromStrings("a", "3"),
+					labels.FromStrings("a", "4"),
+					labels.FromStrings("a", "5"),
+					labels.FromStrings("a", "6"),
+				},
+			},
+			// Due to TSDB compaction delay (not compacting fresh block), we need one more block to be pushed to trigger compaction.
+			{
+				numFloatSamples: 100, mint: 3000, maxt: 4000, extLset: extLabels3, res: 124,
 				series: []labels.Labels{
 					labels.FromStrings("a", "7"),
 				},
 			},
 		})
 
 		require.NoError(t, bComp.Compact(ctx, 0), 0)
-		assert.Equal(t, 5.0, promtest.ToFloat64(sy.metrics.blocksMarkedForDeletion))
+		assert.Equal(t, 7.0, promtest.ToFloat64(sy.metrics.blocksMarkedForDeletion))
 		assert.Equal(t, 0.0, promtest.ToFloat64(metrics.blocksMarkedForNoCompact.WithLabelValues(block.OutOfOrderChunksNoCompactReason)))
 		assert.Equal(t, 0.0, promtest.ToFloat64(sy.metrics.garbageCollectionFailures))
-		assert.Equal(t, 2.0, promtest.ToFloat64(metrics.groupCompactions))
-		assert.Equal(t, 2.0, promtest.ToFloat64(metrics.groupCompactionRunsStarted))
-		assert.Equal(t, 2.0, promtest.ToFloat64(metrics.groupCompactionRunsCompleted))
+		assert.Equal(t, 3.0, promtest.ToFloat64(metrics.groupCompactions))
+		assert.Equal(t, 3.0, promtest.ToFloat64(metrics.groupCompactionRunsStarted))
+		assert.Equal(t, 3.0, promtest.ToFloat64(metrics.groupCompactionRunsCompleted))
 		assert.Equal(t, 0.0, promtest.ToFloat64(metrics.groupCompactionRunsFailed))
-		assert.Equal(t, 2.0, promtest.ToFloat64(metrics.blockUploadsStarted))
+		assert.Equal(t, 3.0, promtest.ToFloat64(metrics.blockUploadsStarted))
```

Comment:
do you know why these changed? it looks like there are more blocks now and more compaction have started after this change

Reply:
Yes , because I've added two new blocks containing native histograms. To test the stats for that as well. See further up in the testcases.

Reply:
i didn't realize this is asserting kind of on the number of test cases above